### ISSUE TITLE
torchscript failed

### ISSUE THREAD

**Issue by ts19970816** [ultralytics/yolov5#4352]
xport: weights=./runs/train/exp16/weights/best.pt, img_size=[640, 640], batch_size=1, device=cpu, include=['torchscript', 'onnx', 'coreml'], half=False, inplace=False, train=False, optimize=False, dynamic=False, simplify=False, opset=12
YOLOv5 🚀 2021-8-3 torch 1.7.1 CPU

Fusing layers... 
Model Summary: 392 layers, 46622106 parameters, 0 gradients, 114.2 GFLOPs

PyTorch: starting from ./runs/train/exp16/weights/best.pt (374.2 MB)

TorchScript: starting export with torch 1.7.1...
/home/ts/project/yolov5-master/models/yolo.py:57: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  if self.grid[i].shape[2:4] != x[i].shape[2:4] or self.onnx_dynamic:
/home/ts/project/yolov5-master/models/yolo.py:57: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  if self.grid[i].shape[2:4] != x[i].shape[2:4] or self.onnx_dynamic:
TorchScript: export failure: save(): incompatible function arguments. The following argument types are supported:
    1. (self: torch._C.ScriptModule, filename: str, _extra_files: Dict[str, str] = {}) -> None

Invoked with: <torch._C.ScriptModule object at 0x7fc81d2dd538>, PosixPath('runs/train/exp16/weights/best.torchscript.pt')
requirements: onnx not found and is required by YOLOv5, attempting auto-update...
requirements: 'pip install onnx' skipped (offline)
requirements: onnx-simplifier not found and is required by YOLOv5, attempting auto-update...
requirements: 'pip install onnx-simplifier' skipped (offline)
ONNX: export failure: No module named 'onnx'

CoreML: export failure: No module named 'coremltools'

Export complete (18.86s)
Results saved to /home/ts/project/yolov5-master/runs/train/exp16/weights
Visualize with https://netron.app

**Comment by fuzzypants123** [comment]
add  f =str(f)  after  f = file.with_suffix('.torchscript.pt')   in function export_torchscript  will be fine.

**Comment by mheriyanto** [comment]
> add f =str(f) after f = file.with_suffix('.torchscript.pt') in function export_torchscript will be fine.

Thanks @fuzzypants123. It works for me.

**Comment by glenn-jocher** [comment]
@ts19970816 @fuzzypants123 @mheriyanto 👋 hi, thanks for letting us know about this possible problem with YOLOv5 🚀. We've created a few short guidelines below to help users provide what we need in order to get started investigating a possible problem.

## How to create a Minimal, Reproducible Example

When asking a question, people will be better able to provide help if you provide **code** that they can easily understand and use to **reproduce** the problem. This is referred to by community members as creating a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). Your code that reproduces the problem should be:

* ✅ **Minimal** – Use as little code as possible that still produces the same problem
* ✅ **Complete** – Provide **all** parts someone else needs to reproduce your problem in the question itself
* ✅ **Reproducible** – Test the code you're about to provide to make sure it reproduces the problem

In addition to the above requirements, for [Ultralytics](https://ultralytics.com/) to provide assistance your code should be:

* ✅ **Current** – Verify that your code is up-to-date with current GitHub [master](https://github.com/ultralytics/yolov5/tree/master), and if necessary `git pull` or `git clone` a new copy to ensure your problem has not already been resolved by previous commits.
* ✅ **Unmodified** – Your problem must be reproducible without any modifications to the codebase in this repository. [Ultralytics](https://ultralytics.com/) does not provide support for custom code ⚠️.

If you believe your problem meets all of the above criteria, please close this issue and raise a new one using the 🐛 **Bug Report** [template](https://github.com/ultralytics/yolov5/issues/new/choose) and providing a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us better understand and diagnose your problem. 

Thank you! 😃
